src/state/rootReducer.js

```diff
@@ -13,15 +13,23 @@ import presentationMode from './presentation_mode/presentationMode.reducer';
 import auth from './auth/auth.reducer';
 
 function createRootReducer(history: History): CombinedReducer<AppState, Action> {
-  return combineReducers({
-    router: connectRouter(history),
-    presentation,
-    activeTeam,
-    teamSidebar,
-    presentationMode,
-    auth,
-    flash: flashReducer,
-  });
+  return (state, action) => {
+    const combinedReducer = combineReducers({
+      router: connectRouter(history),
+      presentation,
+      activeTeam,
+      teamSidebar,
+      presentationMode,
+      auth,
+      flash: flashReducer,
+    });
+
+    const nextState = action.type === 'RESET'
+      ? undefined
+      : state;
```

Comment:
What exactly happens here? If a reset action is sent the state is set to undefined? Will this implicitly log out the user?
since you are logging out on 401 with:
```
      if (response && response.status === 401) {
        store.dispatch({ type: 'RESET' });
      }
```

Or is that unneeded since in a case where users are receiving 401 they are already logged out and their state is just not up to date?

Reply:
It wipes the whole state and sets it to the default values.

Reply:
So it does log out the user too by killing off all the auth information that currently is in the state.
When this auth information is removed the App component automatically leaves the AppAuthenticated and redirects to the LandingPage.

Reply:
Sounds good.